core/src/main/java/com/rudderstack/android/sdk/core/RudderPreferenceManager.java

```diff
@@ -45,6 +48,16 @@ String getTraits() {
         return preferences.getString(RUDDER_TRAITS_KEY, null);
     }
 
+    String getAnonymousId() {
+        return preferences.getString(RUDDER_ANONYMOUS_ID, null);
+    }
+
+    void saveAnonymousId(String anonymousId) {
```

Comment:
Instead of taking the anonymousId from outside, we can add two methods here:

`getAnonymousId()` and `resetAnonymousId()`.

`getAnonymousId` will ensure that you get a nonnull valid string every time. 
`resetAnonymousId` will clear from preferences so that next time you call getAnonymousId you get a new one. 

We can design the entire flow on top of it.

Reply:
changed